File: core/feature10.go
package main

import (
	"math"
)

func feature10(kk, gg, ll, ee, tt, ett float64) (sr, br, sc, bc, zmks, zmgs, wll, dd, a float64) {
	Rad := math.Pi / 180.0
	r := kk / 2.0
	rr := gg / 2.0
	bb := math.Atan((rr-r)/ll) / Rad
	nn := r - ee*math.Sin(bb*Rad)
	mm := rr + ee*math.Sin(bb*Rad)
	uu := 1 / math.Sin(bb*Rad)
	vv := (180.0 / tt) * math.Sin(bb*Rad)
	sr = nn * uu
	br = mm * uu
	sc = 2.0 * nn * uu * math.Sin(vv*Rad)
	bc = 2.0 * mm * uu * math.Sin(vv*Rad)
	vv = (180.0 / ett) * math.Sin(bb*Rad)
	pp := mm*uu - nn*uu
	hh := pp * math.Cos(vv*Rad)
	aa := (mm*uu + nn*uu) * math.Sin(vv*Rad)
	theta := math.Atan(aa/hh) / Rad
	dd = hh * (1.0 / math.Cos(theta*Rad))
	zmks = 2.0 * nn * uu * math.Sin(vv*Rad)
	zmgs = 2.0 * mm * uu * math.Sin(vv*Rad)
	wll = pp
	a = math.Pi * (rr + r) * math.Sqrt(math.Pow(ll, 2.0)+math.Pow(rr-r, 2.0)) * (1.0 / math.Pow(10.0, 6.0))
	return
}

File: core/feature10.js
function feature10() {
  var parameters = extractParameters(["K", "G", "L", "E", "T", "ET"]);
  return feature10_core(parameters);
}

function feature10_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var K = getParameter(parameters, "K");
  var G = getParameter(parameters, "G");
  var L = getParameter(parameters, "L");
  var E = getParameter(parameters, "E");
  var T = getParameter(parameters, "T");
  var ET = getParameter(parameters, "ET");
  var result = {};
  var Rad = Math.PI / 180.0;
  var r = K / 2.0;
  var R = G / 2.0;
  var B = Math.atan((R - r) / L) / Rad;
  var N = r - E * Math.sin(B * Rad);
  var M = R + E * Math.sin(B * Rad);
  var U = 1 / Math.sin(B * Rad);
  var V = (180.0 / T) * Math.sin(B * Rad);
  result["SR"] = N * U;
  result["BR"] = M * U;
  result["SC"] = 2.0 * N * U * Math.sin(V * Rad);
  result["BC"] = 2.0 * M * U * Math.sin(V * Rad);
  V = (180.0 / ET) * Math.sin(B * Rad);
  var P = M * U - N * U;
  var H = P * Math.cos(V * Rad);
  var A = (M * U + N * U) * Math.sin(V * Rad);
  var theta = Math.atan(A / H) / Rad;
  result["D"] = H * (1.0 / Math.cos(theta * Rad));
  result["ZMKS"] = 2.0 * N * U * Math.sin(V * Rad);
  result["ZMGS"] = 2.0 * M * U * Math.sin(V * Rad);
  result["WLL"] = P;
  result["Area"] = Math.PI * (R + r) * Math.sqrt(Math.pow(L, 2.0) + Math.pow(R - r, 2.0)) * (1.0 / Math.pow(10.0, 6.0));
  return result;
}
